nixos/modules/installer/tools/nixos-version.sh

```diff
@@ -1,23 +1,36 @@
 #! @runtimeShell@
 
+if [ $# -eq 0 ]; then
+    echo "@version@ (@codeName@)"
+    exit 0
+fi
+
 case "$1" in
   -h|--help)
     exec man nixos-version
     exit 1
     ;;
   --hash|--revision)
     if ! [[ @revision@ =~ ^[0-9a-f]+$ ]]; then
-      echo "$0: Nixpkgs commit hash is unknown"
+      echo "$0: Nixpkgs commit hash is unknown" >&2
       exit 1
     fi
     echo "@revision@"
     ;;
+  --configurationRevision)
+    if ! [[ @configurationRevision@ =~ ^[0-9a-f]+$ ]]; then
```

Comment:
Not quite the right check, since the configurationRevision could be set to arbitrary text (including uppercase letters). Maybe check if empty instead?